src/ScribuntoLuaLibrary.php

```diff
@@ -93,4 +106,118 @@ public function getPropertyType( $propertyName = null ) {
 		return array( $property->findPropertyTypeID() );
 	}
 
+	/**
+	 * This mirrors the functionality of the parser function #info and makes it available to lua.
+	 *
+	 * @param string	$text	text to show inside the info popup
+	 * @param string    $icon   identifier for the icon to use
+	 *
+	 * @uses \SMW\ParserFunctionFactory::__construct
+	 *
+	 * @return string[]
+	 */
+	public function info( $text, $icon='info' )
+	{
+		$parser = $this->getEngine()->getParser();
+
+		# do some parameter processing
+		if ( !trim($text) || ! is_string($text) ) {
```

Comment:
Rather than writing comments (for which we use `//`), consider splitting up the function into smaller, well named functions, that do one single thing. That makes the code a lot easier to read and maintain.